tests/test_widget.py

```diff
@@ -317,3 +316,76 @@ def check_dialog_and_click_no(dialog):
         dialog_action=check_dialog_and_click_no,
         time_out=5,
     )
+
+
+def test_radius_save_preserves_old_data(cpt_widget):
+    """Test saving a new particle radius does not mess up the previous one."""
+    # setup measurement layer
+    # layer_measurements = cpt_widget._setup_measurement_layer()
+    layer_measurements = cpt_widget.viewer.add_points(name="Radii and Lengths", ndim=2)
+
+    # add a new particle and calculate radius
+    cpt_widget.particle_decays_menu.setCurrentIndex(1)
+    # layer_measurements.add([[0, 0, 0, 1], [0, 0, 1, 0], [0, 0, 0, -1]]) # 1px radius
+    layer_measurements.add([[0, 1], [1, 0], [0, -1]])  # 1px radius
+    layer_measurements.selected_data = {0, 1, 2}
+    cpt_widget._on_click_radius()
+
+    # create a second particle and calculate radius
+    cpt_widget.particle_decays_menu.setCurrentIndex(1)
+    # layer_measurements.add([[0, 0, -6, 3], [0, 0, -3, 2], [0, 0, 0, 3]]) # 5px radius
+    layer_measurements.add([[-6, 3], [-3, 2], [0, 3]])  # 5px radius
+    layer_measurements.selected_data = {3, 4, 5}
+    cpt_widget._on_click_radius()
+
+    # check the radius information is different
```

Comment:
```suggestion
```

And here, you don't need a comment because you have the assertion messages saying what you're checking.